back/src/database/makeSQL.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const SearchWords = require("../objects/SearchWords");
+
+module.exports.getSpotQueryBuilder = function(keywords){
+    //input : keyword is Json
+    //{"spotId" : "sample"} 
+    var query = 'SELECT * FROM spots.spots';
+    var andFlag = false;//for only one keyword
+    var where = []
+    if( keywords.spotId != null )
+        where.push(` spot_id='${keywords.spotId}'`);
+    if( keywords.spotName != null )
+        where.push(` spot_name='${keywords.spotName}'`);
+    if( keywords.geom != null )
+        where.push(` geom='${keywords.geom}'`);
+    if( keywords.spotType != null )
+        where.push(` spot_type='${keywords.spotType}'`);
+    if( keywords.userId != null )
+        where.push(` user_id='${keywords.userId}'`);
+
+    if(where.length!=0){
+        query += ' where';
+        query += where.join(' and ');
+    }
+    query += ';';
+    return query;
+}
+
+module.exports.getReviewsQueryBuilder = function(spotIds){
+    var query = `SELECT * FROM spots.review where `;
+    if(spotIds == null){
```

Comment:
if (array === undefined || array.length == 0) {
    // array empty or does not exist
}
This checks if list is defined and if there are ids in the list